spyder/plugins/explorer/widgets/explorer.py

```diff
@@ -1563,8 +1578,20 @@ def update_history(self, directory):
             user_directory = get_home_dir()
             self.chdir(directory=user_directory, browsing_history=True)
 
-    def chdir(self, directory=None, browsing_history=False):
-        """Set directory as working directory"""
+    def chdir(self, directory=None, browsing_history=False, emit=True):
+        """
+        Set directory as working directory.
+
+        Parameters
+        ----------
+        directory: str
+            The new working directory.
+        browsing_history: bool, optional
+            Add the new `directory`to the browsing history. Default is False.
+        emit: bool, optional

```

Comment:
The plugin is now correctly decoupled (removed the refresh_explorer etc params) so to avoid having an endless cycle of working directory updates if `plugin A` updates the working dir, then the working directory plugin updates `plugin B` but tells it to NOT emit a signal that it has updated the working dir.

Reply:
Ok, understood.